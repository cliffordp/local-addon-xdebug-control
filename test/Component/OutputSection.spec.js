const React = require( 'react' )
const Component = require( 'react' ).Component
const renderer = require( 'react-test-renderer' )

const context = {
	'React': React,
}

const OutputSection = require( './../../src/Component/OutputSection' )( context )

describe( '<OutputSection/>', function () {
	const defaultProps = {
		children: ['foo', 'bar', 'baz'].map( ( e ) => {
			return (
				<p>{e}</p>
			)
		} ),
	}

	it( 'renders correctly with props', function () {
		const component = renderer.create( <OutputSection {...defaultProps}/> )
		let tree = component.toJSON()

		expect( tree ).toMatchSnapshot()
	} )

	it( 'renders correctly without children', function () {
		const props = Object.assign( {}, defaultProps, {children: []} )
		const component = renderer.create( <OutputSection {...defaultProps}/> )
		let tree = component.toJSON()

		expect( tree ).toMatchSnapshot()
	} )
} )
