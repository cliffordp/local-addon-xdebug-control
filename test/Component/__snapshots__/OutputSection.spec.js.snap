// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OutputSection/> renders correctly with props 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <div
    style={
      Object {
        "marginBottom": ".5em",
        "textAlign": "center",
      }
    }
  >
    <p>
      foo
    </p>
  </div>
  <div
    style={
      Object {
        "marginBottom": ".5em",
        "textAlign": "center",
      }
    }
  >
    <p>
      bar
    </p>
  </div>
  <div
    style={
      Object {
        "marginBottom": ".5em",
        "textAlign": "center",
      }
    }
  >
    <p>
      baz
    </p>
  </div>
</div>
`;

exports[`<OutputSection/> renders correctly without children 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <div
    style={
      Object {
        "marginBottom": ".5em",
        "textAlign": "center",
      }
    }
  >
    <p>
      foo
    </p>
  </div>
  <div
    style={
      Object {
        "marginBottom": ".5em",
        "textAlign": "center",
      }
    }
  >
    <p>
      bar
    </p>
  </div>
  <div
    style={
      Object {
        "marginBottom": ".5em",
        "textAlign": "center",
      }
    }
  >
    <p>
      baz
    </p>
  </div>
</div>
`;
